sys/shell/commands/sc_cord_ep.c

```diff
@@ -63,6 +63,10 @@ int _cord_ep_handler(int argc, char **argv)
         }
         if (argc > 3) {
             regif = argv[3];
+            if (regif[0] != '/') {
```

Comment:
Perhaps it makes sense just to change the command and accept an URI, which we parse with https://doc.riot-os.org/group__sys__uri__parser.html ?

Reply:
Hm. A reasonable suggestion but should the shell command enforce that the `registration interface` is a valid uri? 
At the moment the chain of responsibility is: 
1. `registration interface` -> defined somewhere in some coap related rfc
2. `gcoap_req_init()` -> uses a `registration interface`
3. `gcoap_req_init()` -> enforces some bare minimal expectation how a  `registration interface` should look like (mostly for programming safety I assume, didn't look deeply at it)
4. `cord_ep_register()` -> passes a `registration interface`(without any checks on it, beside a convenience NULL handling) to `gcoap_req_init()`
5. `_cord_ep_handler()` -> calls `cord_ep_register()` with a user given `registration interface` (without any checks on it)
6. `the user` -> "Generates" a `registration interface` and calls `_cord_ep_handler()` with it (maybe with checks, maybe without, users tend to be unreliable 😬)

Given this chain & your proposal, I think I could look up in coap / the rfc how exactly a `registration interface` may and or *must* look like (URI...). After that is clarified (at least for me :p) one should enforce that as close to the rfc as possible (in the chain).  One aspect that is important for deciding this is performance / code size etc. E.g. a heavy URI check may not be wanted in 3. gcoap, assuming that only other software interacts with it(but not users). This would delegate that responsibility to 4. `cord_ep_register()`.
Enforcing the `registration interface` to be compliant with a given uri parser inside the shell command, while the e.g. `cord_ep_register()` could perfectly work with a non uri compliant `registration interface`, could become confusing. 

E.g. A user could find that a given `registration interface` works in his/her own code but not in the shell. This is also the reason why I tried to match the assert as close as possible and didn't add any other checks.

Reply:
> Given this chain & your proposal, I think I could look up in coap / the rfc how exactly a `registration interface` may and or _must_ look like (URI...). After that is clarified (at least for me :p) one should enforce that as close to the rfc as possible (in the chain). One aspect that is important for deciding this is performance / code size etc. E.g. a heavy URI check may not be wanted in 3. gcoap, assuming that only other software interacts with it(but not users). This would delegate that responsibility to 4. `cord_ep_register()`. Enforcing the `registration interface` to be compliant with a given uri parser inside the shell command, while the e.g. `cord_ep_register()` could perfectly work with a non uri compliant `registration interface`, could become confusing.
> 
> E.g. A user could find that a given `registration interface` works in his/her own code but not in the shell. This is also the reason why I tried to match the assert as close as possible and didn't add any other checks.

As I understand this will always end up being a valid URI. We need to get from the command the RD address (maybe port) pointing to a CoAP endpoint, and CoAP path (i.e. the interface). I think unifying the shell commands through our examples is a big gain, then we don't re-implement parsers each time. My proposal is to merge the argument 2 and 3 (remote and interface) into a single URI.